src/sync_template.sh

```diff
@@ -243,11 +251,30 @@ function pull_source_changes() {
 
   eval "git pull ${source_repo} --tags ${git_remote_pull_params}" || pull_has_issues=true
 
+  info "finished pulling from the source."
+  info "logging out from source ${SOURCE_REPO_HOSTNAME}."
+
+  if [[ -n "${SRC_SSH_PRIVATEKEY_ABS_PATH}" ]] &>/dev/null; then
+    info "we are using ssh for the source repo. No need to logout."
+  else
+    gh auth status --hostname "${SOURCE_REPO_HOSTNAME}" 
+    gh auth status --active    
+
+    # Clear the GH_TOKEN environment variable
+    unset GH_TOKEN
```

Comment:
hmm not super sure why we need to unset this env variable. If needed, then possibly it is sufficient to rename the variable within `action.yml`.